hcsctl/pkg/rook/issue-template.go

```
@@ -0,0 +1,521 @@
+		}
+	}
+
+	return failMessage
+}
+
+func createInfoDir(dirPath, dirName string) (string, error) {
+	infoDir := path.Join(dirPath, dirName)
+	err := os.MkdirAll(infoDir, 0755)
+
+	if err != nil {
+		err = errors.Wrap(err, "fail to create directory "+dirName)
+	}
+
+	return infoDir, err
+}
+
+func writeInfo(infoDir, command, infoName string, cmd []string) error {
+	var stdout, stderr bytes.Buffer
+
+	var err error
+
+	message := "[Run][" + command + "] infoName: " + infoName + ", cmd: [" + strings.Join(cmd, " ") + "]"
+	glog.Info(message)
+
+	switch command {
+	case kubeCommand:
+		err = kubectl.Run(&stdout, &stderr, cmd...)
+	case cephCommand:
+		stdout, err = execInToolbox(&stderr, cmd...)
+	case commonCommand:
+		err = executeCommand(&stdout, &stderr, cmd[0], cmd[1:]...)
+	default:
+		err = errors.New("ERROR: unknown command")
+	}
+
+	if err != nil {
+		errorMessage := "[Fail][" + command + "] infoName: " + infoName +
+			", cmd: [" + strings.Join(cmd, " ") + "], message: " + stderr.String()
+		err = errors.Wrap(err, errorMessage)
+
+		return err
+	}
+
+	return writeInfoFile(infoDir, infoName, stdout.String())
+}
+
+func writeInfoFile(infoDir, infoName, output string) error {
+	filePath := path.Join(infoDir, infoName)
+	f, err := os.Create(filePath)
+
+	if err != nil {
+		err = errors.Wrap(err, "fail to write "+infoName)
+		return err
+	}
+
+	defer deferCheck(f.Close)
+
+	_, err = f.WriteString(output)
+
+	if err != nil {
+		err = errors.Wrap(err, "fail to write "+infoName)
+	}
+
+	return err
+}
+
+func zip(source, target string) error {
+	filename := filepath.Base(source)
+	target = filepath.Join(target, fmt.Sprintf("%s.tar.gz", filename))
+	writer, err := os.Create(target)
+
+	if err != nil {
+		return err
+	}
+
+	defer deferCheck(writer.Close)
+
+	archiver := gzip.NewWriter(writer)
+	defer deferCheck(archiver.Close)
+
+	tarball := tar.NewWriter(archiver)
+	defer deferCheck(tarball.Close)
+
+	info, err := os.Stat(source)
+	if err != nil {
+		return nil
```

why does it return `nil`? zip function is called after creating `source` directory (`rookInfoDirName`), so if any error occurs here, it should return that error to caller.